rs/nervous_system/tools/sync-with-released-nervous-system-wasms/src/main.rs

```diff
@@ -52,6 +69,95 @@ async fn get_mainnet_canister_git_commit_id_and_module_hash(
     Ok((git_commit_id, module_hash))
 }
 
+async fn get_mainnet_canister_git_tag_and_module_hash(
+    agent: &Agent,
+    canister_id: CanisterId,
+    canister_repository: String,
+    canister_name: String,
+) -> Result<(String, String)> {
```

Comment:
I would be tempted to ask some AI assistant to generate a test for this function.

Reply:
This function will be tested by being used by the NNS bot, i.e., if it doesn't work, then the PR created by the bot will fail, right?

Reply:
> If the bot doesn't create a PR when it should (and doesn't produce any visible errors), how would we see that?

In that case, the CI workflow to update mainnet canister versions should fail.

Reply:
> If the bot doesn't see any changes, it's not going to fail, it's going to assume that the pinned versions are still up to date. How would we distinguish that from a bug in the bot?

To address this concern, I'd suggest to always recreate the entire json file from scratch instead of only updating it. This forces the bot to actively look over all canisters.